azure/services/aso/aso.go

```diff
@@ -342,5 +326,5 @@ func (r *reconciler[T]) PauseResource(ctx context.Context, spec azure.ASOResourc
 	annotations[asoannotations.ReconcilePolicy] = string(asoannotations.ReconcilePolicySkip)
 	resource.SetAnnotations(annotations)
 
-	return helper.Patch(ctx, resource)
+	return r.Client.Patch(ctx, resource, client.MergeFrom(before))
```

Comment:
What is the difference between `r.Client.Patch()` and `helper.Patch()`? I'm assuming that from the comment above that we no longer need the two behaviors you listed [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4497/files#r1462264117).

Reply:
The helper wraps the client `Patch` in a bunch of other stuff that I don't think we need here like you saw in my other comment. Ultimately I expect the same API calls to result from this change.